File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.cpp
#include<bits/stdc++.h>
using namespace std;

int ret = 0;
int solution(int n) {
    for(int i = 1; i <= n; i++){
        int sum = 0;
        for(int j = i; j <= n; j++){
            sum += j;
            if(sum == n){ret++; break;}
            else if(sum > n) break;
        }
    }
    return ret;
}

File: Algorithms/프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int ret = 1;
        
        for(int i = 1; i <= n / 2; i++){
            int sum = 0;
            for(int j = i; j < n; j++){
                sum += j;
                if(sum == n){
                    ret++; break;
                }else if(sum > n) break;
            }
        }
        
        return ret;
    }
}